### PR TITLE
Replace is_zero_approx(A.distance_to(B)) with A==B

### PR DESCRIPTION
**Pull request by ptrojahn**
Related to #22988 (Fixes the holes in the shape of
the first comment)

### CODE REVIEW

File: modules/csg/csg.cpp [hunk 1]
```diff
@@ -242,7 +242,7 @@ void CSGBrushOperation::BuildPoly::_clip_segment(const CSGBrush *p_brush, int p_
 	//check if edge and poly share a vertex, of so, assign it to segment_idx
 	for (int i = 0; i < points.size(); i++) {
 		for (int j = 0; j < 2; j++) {
-			if (Math::is_zero_approx(segment[j].distance_to(points[i].point))) {
+			if (segment[j].distance_to(points[i].point) < 0.00005) {
```
Comment (Calinou) [inline modules/csg/csg.cpp]: There should probably be a comment that justifies the magic number's presence :slightly_smiling_face:

Alternatively, could `CMP_EPSILON * 2` be used here?

Comment (ptrojahn) [reply modules/csg/csg.cpp]: Not sure if 0.00005 is large enough. It's enough to fix the example in the issue, but maybe there are other cases where it isn't.

Comment (fire) [reply modules/csg/csg.cpp]: Can you investigate is_equal_approx_ratio?

**Comment by aaronfranke** [comment]
The reason this issue exists is that `is_zero_approx` is not actually the right thing to use here in the first place. Increasing the tolerance will only hide the true cause of the problem.

The purpose of this code is seems to be to check if the distance between two vectors is nearly zero. In other words, check if the vectors are approximately equal. Thanks to #18992, this is [what the == operator does](https://github.com/godotengine/godot/blob/master/core/math/vector2.h#L215), so the fix should be `if (segment[j] == points[i].point) {`

**Comment by ptrojahn** [comment]
@aaronfranke  Good idea! Didn't know == handles that.

**Review by aaronfranke — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**